### PR TITLE
Fix that execution errors of a term are swallowed when there are functions before the term

### PR DESCRIPTION
**Pull request by chengluyu**

### CODE REVIEW

File: shared/src/test/scala/mlscript/DiffTests.scala [hunk 1]
```diff
@@ -258,7 +258,7 @@ class DiffTests
           case "dv" => mode.copy(debugVariance = true)
           case "ge" => mode.copy(expectCodeGenErrors = true)
           case "re" => mode.copy(expectRuntimeErrors = true)
-          case "r" | "showRepl" => mode.copy(showRepl = true)
+          case "ShowRepl" => mode.copy(showRepl = true)
```
Comment (LPTK) [inline shared/src/test/scala/mlscript/DiffTests.scala]: Why would you do that? There's a reason I changed it to lower case... Only commands that modify the global file state are upper case!

Comment (chengluyu) [reply shared/src/test/scala/mlscript/DiffTests.scala]: This is not a deliberate change. It was accidentally caused when I switched branches.

**Comment by LPTK** [comment]
Thanks! But it looks like there are uncommited diff-test changes. And you left debugging comments (the "what" things).

**Comment by chengluyu** [comment]
It's removed!

**Review by LPTK — CHANGES_REQUESTED** [review]
Thanks but please read https://github.com/hkust-taco/mlscript/blob/mlscript/CONTRIBUTING.md again and try to avoid wasting my time too much 😓

**Comment by LPTK** [comment]
@chengluyu Let me just be clear here: what upsets me is when someone asks for a review on their PR and it turns out they haven't even reviewed their own diff before doing so. **Please review your own diff.**

File: shared/src/test/diff/codegen/Mixin.mls [hunk 2]
```diff
@@ -256,7 +256,7 @@ TestLang.eval(mk(0))
 //│ res = TestLang.eval(mk(0));
 //│ // End of generated code
 //│ res
-//│     = [Function: mk]
```
Comment (chengluyu) [inline shared/src/test/diff/codegen/Mixin.mls]: I guess the `:js` flag for this block is to see why `res` is `mk`. Right? If so, I will remove `:js` flag. P.S. Expand to see the `:js` flag above.

**Review by chengluyu — COMMENTED** [review]
I have read all changes. I left one comment. Please have a look.

File: shared/src/test/diff/codegen/NuFuns.mls [hunk 3]
```diff
@@ -26,7 +26,7 @@ foo
 //│ fun foo: Int
 //│ Int
 //│ res
-//│     = [Function: foo1]
+//│     = 111
 
 
 // FIXME: returns a thunk
```
Comment (LPTK) [inline shared/src/test/diff/codegen/NuFuns.mls]:
```suggestion
```

File: shared/src/test/diff/codegen/Mixin.mls (hunk 2 above)
Comment (LPTK) [reply shared/src/test/diff/codegen/Mixin.mls]: Yeah probably. Let's remove it!

**Review by LPTK — APPROVED** [review]
Thanks. When the comments are addressed, feel free to go ahead and squash-merge this (with a clean commit message).

**Pull request merged**